pkg/cloud/services/compute/instance.go

```diff
@@ -519,7 +524,9 @@ func deletePorts(s *Service, nets []servers.Network) error {
 				return nil
 			}
 		}
-		if err := ports.Delete(s.networkClient, n.Port).ExtractErr(); err != nil {
+		mc := metrics.NewMetricPrometheusContext("port", "delete")
+		err = ports.Delete(s.networkClient, n.Port).ExtractErr()
+		if mc.ObserveRequest(err) != nil {
```

Comment:
An alternative here would be the following to keep the `if err :=` (same for other occurencies):
```suggestion
		mc := metrics.NewMetricPrometheusContext("port", "delete")
		if err := ports.Delete(s.networkClient, n.Port).ExtractErr()
; mc.ObserveRequest(err) != nil {
```

But also ok for me as is :-)